pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -58,18 +58,18 @@ var gvk = schema.GroupVersionKind{
 }
 
 func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	if createValidation != nil {
-		if err := createValidation(obj.DeepCopyObject()); err != nil {
-			return nil, err
-		}
-	}
-
 	out := obj.(*authenticationapi.TokenRequest)
 
 	if errs := authenticationvalidation.ValidateTokenRequest(out); len(errs) != 0 {
 		return nil, errors.NewInvalid(gvk.GroupKind(), "", errs)
 	}
 
+	if createValidation != nil {
+		if err := createValidation(obj.DeepCopyObject()); err != nil {
```

Comment:
for validation, why do we need to copy ?

Reply:
to ensure the object is not modified by validation. this matches the way the generic registry calls createValidation